### DISCUSSION TITLE
Some Concerns regarding ArchInstall

### DISCUSSION THREAD

**Discussion by DarkXero-dev** [archlinux/archinstall#2361] [General]
Hello..

I am just cross-posting this here from Arch Forums...

After 3 years of maintaining my own custom Arch Distro XeroLinux, I have recently decided to drop it in favor of Vanilla Arch using the infamous ArchInstall script.. Not gonna discuss the reasons as it's not what is important. Which brings me to the subject at hand. Keep in mind that I am looking at it from the perspective of an ex Distro Maintainer..

On with my concerns...

First off let me discuss Grub bootloader :

I have noticed that when it's selected, it does not include the --bootloader-id=insertnamehere which would add the necessary files in /boot or /boot/efi. All we get with this script is /boot/BOOTx86-64.efi for whatever reason. Also we don't get all the necessary hooks that allow to reinstall it making sure it stays functional. Albeit not necessary but would prove to be useful. Ignore this part if am wrong.

Second concern, nVidia Drivers :

If one has this GPU (16 series and up) and selects it during install with a Wayland based WM or wants to use Wayland at some point, all the installer does is install a few driver packages not all of them nor does it do the necessary modifications like mkinitcpio.conf modules and Grub bootloader flags which blacklist nouveau and early load of the nvidia.drm modules. IMO that is not ideal for people who select Hyprland for example and might end up with a less than ideal experience having to do all that manually post-install.

Third concern Profiles :

I have tested a few, but since I mainly daily drive KDE Plasma, I noticed that the profile is bare and too minimal. Might I suggest you include an option to select either minimal or full ? Coz not having Spectacle, gwenview, dolphin-plugins and other useful packages is a pain having to install them ourselves post-install. (optional though useful).

Finally multilib repo :

IMO that repo should be enabled by default as I have noticed a lot of users forget to enable it resulting in a lot of issues installing apps that require it like Steam etc..

Also just noticed linux-headers not included but needed.

Please take some of my concerns with a grain of salt just looking at things differently from other users..

Thanks. Let me know what you think ;)

**Comment by codefiles** [comment]
> I have noticed that when it's selected, it does not include the --bootloader-id=insertnamehere which would add the necessary files in /boot or /boot/efi.

That option is included.

https://github.com/archlinux/archinstall/blob/f107104103b74ac065a56bd94a0878f0d263e177/archinstall/lib/installer.py#L954

> IMO that repo should be enabled by default as I have noticed a lot of users forget to enable it resulting in a lot of issues installing apps that require it like Steam etc..

The Arch Linux developers do not have it as a default in pacman, why would you think they would approve of it as a default in archinstall?

> Also just noticed linux-headers not included but needed.

Why do you think this is needed? Certainly all use cases do not require this.

**Comment by DarkXero-dev** [comment]
> That option is included.
![image](https://github.com/archlinux/archinstall/assets/81101439/fd9969ae-ba7c-48ee-9ad9-5fce4253b42b)

![image](https://github.com/archlinux/archinstall/assets/81101439/c8e7653c-3edb-4138-a20e-8450b100ac8a)

> The Arch Linux developers do not have it as a default in pacman, why would you think they would approve of it as a default in archinstall?

Well if user selects something like drivers that requires 32bit libraries and forgets to enable it might end up with issues. Safer to enable...

> Why do you think this is needed? Certainly all use cases do not require this.

Well coz it's required by many things, like nVidia drivers, doesn't hurt to include it by default uses very little space and will not break anything..

**Reply by 413x1nkp** [reply]
> Well if user selects something like drivers that requires 32bit libraries and forgets to enable it might end up with issues. Safer to enable...

You are right about that, however, due to original design, Arch is meant to be user-centric; users only install what THEY need.
Figuring out for a user what exactly they need is something that lies on the shoulders of the user, this is just how it is designed.

Regarding 32bit libs: If you try to install a package that depends specifically on 32bit lib packages you'll receive messages in the terminal about the libs being not found. After that - a quick google search is all that is needed to fix the issue.
The real issue is that if a user is too hesitant on even googling about the issue - it's on them, not on the developers.

> Well coz it's required by many things, like nVidia drivers, doesn't hurt to include it by default uses very little space and will not break anything..

The issue is that most 32bit libraries can be used instead of x86_64 libraries for a lot of packages, therefore, to avoid user's confusion and unintended behavior, it's best to leave the choice to the users themselves.

**Reply by DarkXero-dev** [reply]
Well I disagree. There is absolutely no scenario where that repo isn't needed. In every one you can think of it is... Steam, Drivers, apps... Not having it enabled by default is just wrong... Coz users will not remember to enable it. Seen situation happen many times.

But hey it is what it is. I have learned that ArchInstall is never gonna change an iota... I said what needed to be said. My job here is done. It's dev's decision in the end.

Just know that everyone I talked to agrees with me 100%.. And profiles need serious rework.. Again many people agree with that too, but whatever... Nothing will change. I as a G33k created my own script bypassing all them limitations. Not everyone will though...
